controller_manager:
  ros__parameters:
    update_rate: 400 # Hz

    joint_state_broadcaster:
      type: joint_state_broadcaster/JointStateBroadcaster
    joint_trajectory_controller:
      type: joint_trajectory_controller/JointTrajectoryController
    
    imu_sensor_broadcaster:
      type: imu_sensor_broadcaster/IMUSensorBroadcaster

    effort_controllers:
      type: effort_controllers/JointGroupEffortController

    quadruped_controller:
      type: quadruped_controller/QuadrupedController

imu_sensor_broadcaster:
  ros__parameters:
    sensor_name: imu
    frame_id: imu_link
    publish_rate: 100.0




joint_state_broadcaster:
  ros__parameters:
    publish_rate: 1000.0
    

quadruped_controller:
    ros__parameters:
      leg_names:
        - front_left
        - front_right
        - rear_left
        - rear_right

      joints:
        - front_left_first_joint
        - front_left_second_joint
        - front_left_third_joint
        - front_right_first_joint
        - front_right_second_joint
        - front_right_third_joint
        - rear_left_first_joint
        - rear_left_second_joint
        - rear_left_third_joint
        - rear_right_first_joint
        - rear_right_second_joint
        - rear_right_third_joint

      command_interface: acceleration
      state_interfaces:
        - position
        - velocity
        - effort

      # MOCK
      # first_joint_kp: 0.5633300297324
      # first_joint_kd: 1.02549019607843
      # second_joint_kp: 0.5633300297324
      # second_joint_kd: 1.02549019607843
      # third_joint_kp: 0.5633300297324
      # third_joint_kd: 1.02549019607843

      # REAL ROBOT
      # 22.5633300297324 * 0.0102549019607843
      first_joint_kp: 22.5633300297324
      first_joint_kd: 0.23138473736372606
      second_joint_kp: 22.5633300297324
      second_joint_kd: 0.23138473736372606
      third_joint_kp: 22.5633300297324
      third_joint_kd: 0.23138473736372606

      # Mujoco:
      # 95.92020756982738, 51.207090397090106, 51.207090397090106,

      # 95.92020756982738 * 0.022526014640095713,
      # 51.207090397090106 * 0.023184721126433695,
      # 51.207090397090106 * 0.023184721126433695,
